contracts/lib/OnChainIdentity.sol

```diff
@@ -0,0 +1,275 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.16;
+pragma abicoder v2;
+
+import "../interfaces/IState.sol";
+import "../lib/SmtLib.sol";
+import "../lib/Poseidon.sol";
+import "../lib/GenesisUtils.sol";
+
+// /**
+//  * @dev Contract managing onchain identity
+//  */
+library OnChainIdentity {
+    using SmtLib for SmtLib.Data;
+
+    /**
+     * @dev Identity data
+     * Issuer Id
+     * Identity identifier
+     * Identity state
+     * State contract
+     * Mapping of roots by state
+     */
+    struct IdentityData {
+        uint256 identityId;
```

Comment:
No need in prefix `identity` in `identityId` and `identityLatestState`. You are already inside `IdentityData`. Just `id` and `latestState`.